File: Models/StationModel.cs
namespace MirekInzFinal.Models
{
    public class StationModel
    {
        public string Name { get; set; } = string.Empty;

        public List<ReservationDayModel> ReservationDays { get; set; } = new List<ReservationDayModel>();

        public StationModel(string name, List<ReservationDayModel> reservationDays)
        {
            Name = name;
            ReservationDays = reservationDays;
        }

        public StationModel()
        {

        }
    }
}

File: ClientApp/src/app/Models/station.model.ts
import {ReservationDayModel} from "./reservation-day.model";

export class StationModel {
  public name: string;
  public reservationDays: ReservationDayModel[]

  constructor(name: string, reservationDays: ReservationDayModel[]) {
    this.name = name;
    this.reservationDays = reservationDays;
  }
}
